faucet-server/Dockerfile

```diff
@@ -0,0 +1,50 @@
+#you can set the Swift version to what you need for your app. Versions can be found here: https://hub.docker.com/_/swift
+FROM swift:5.0.1-xenial as builder
+
+# For local build, add `--build-arg env=docker`
+# In your application, you can use `Environment.custom(name: "docker")` to check if you're in this env
+ENV DEBIAN_FRONTEND noninteractive
+ARG env
+
+RUN apt-get -qq update && apt-get -q -y install \
+  pkg-config libssl-dev wget libsodium-dev software-properties-common libsqlite3-dev \
+  tzdata \
+  && rm -r /var/lib/apt/lists/*
+
+RUN wget https://download.libsodium.org/libsodium/releases/libsodium-1.0.16.tar.gz \
+    && tar xf libsodium-1.0.16.tar.gz && cd libsodium-1.0.16 \
+    && ./configure && make -j4 && make install \
+    && ldconfig
+
+WORKDIR /app
+COPY . .
+RUN mkdir -p /build/lib && cp -R /usr/lib/swift/linux/ /build/lib/
+RUN swift build -c release && mv `swift build -c release --show-bin-path` /build/bin
+# Production image
+FROM ubuntu:16.04
+ARG env
+RUN apt-get -qq update && apt-get install -y \
+  nginx build-essential wget libicu55 libxml2 libbsd0 libcurl3 libatomic1 \
+  tzdata \
+  && rm -r /var/lib/apt/lists/*
+RUN wget https://download.libsodium.org/libsodium/releases/libsodium-1.0.16.tar.gz \
```

Comment:
Why do I need to download `libsodium` repeatedly because the copy is invalid?

`COPY --from=builder /build/lib/* /usr/lib/`

Reply:
This just copy from under, I should have a try to reply you.

Reply:
`wget` is download

Will it use the previous cache?

Reply:
![image](https://user-images.githubusercontent.com/23251844/57356421-faed3100-71a2-11e9-97df-f5c55eb1af75.png)
Sure.

Reply:
Now I will delete this to retry.

Reply:
I tested it, it only uses its own cache, and does not use the `FROM swift:5.0.1-xenial` cache

Reply:
You can try to modify the scripts on lines 14 and 30 and rebuild the image to test the issue.

Reply:
So when you first build the image, it will be downloaded twice.

Reply:
There is no doubt about this, `From swift:5` and `From:Ubuntu:16.04` is different image, even download twice , one in swift5 image , one in our custom image witch we use, so there is no influence to our image.

Reply:
And now, I can delete it. 
Because our image needn't download `libsodium` again.